Sidebar_Chart_Suggestions.gs

```diff
@@ -0,0 +1,252 @@
+function onOpen() {
+  SpreadsheetApp.getUi() 
+      .createMenu('Custom Menu')
+      .addItem('Create', 'showSidebar')
+      .addToUi();
+}
+
+function showSidebar() {
+  var html = HtmlService.createHtmlOutputFromFile('Page')
+      .setTitle('Present dataset')
+      .setWidth(300);
+  SpreadsheetApp.getUi() 
+      .showSidebar(html);
+}
+
+function newChart(range_ip, chartNumber) {
+  var sheet = SpreadsheetApp.getActiveSheet(); 
+  var range = sheet.getRange(range_ip); 
+  var chartBuilder = sheet.newChart();
+  chartBuilder.addRange(range)
+      .setPosition(5, 5, 0, 0)
+     
+  if(chartNumber==0){
+    chartBuilder.setChartType(Charts.ChartType.LINE).setOption('title', 'Line Chart');
+  }
+  else if(chartNumber==1){
+    chartBuilder.setChartType(Charts.ChartType.BAR).setOption('title', 'Bar Chart');
+  }
+  else if (chartNumber==2){
+    chartBuilder.setChartType(Charts.ChartType.COLUMN).setOption('title', 'Column Chart');
+  }
+  else if (chartNumber==3){
+    chartBuilder.setChartType(Charts.ChartType.PIE).setOption('title', 'Pie Chart');
+  }
+  sheet.insertChart(chartBuilder.build());
+}
+
+//merge isNumber and isString functions into isPrimitive() 
+function isNumberRow(data, row_idx){ 
+  Logger.log('number row'+row_idx); 
+  var count = 0;
+  var row_length = data[row_idx].length;
+  for(var j=0; j<row_length; j++){ 
+      if(typeof(data[row_idx][j])==='undefined'){
+         return false;
+      }
+      Logger.log(typeof(data[row_idx][j]));
+      if(typeof(data[row_idx][j])==='number'){
+          count++;   
+      }
+      Logger.log(count);
+      if(count>=10 || count==row_length){
+          return true;
+      }
+  }  
+  return false;
+}
+
+function isNumberCol(data, col_idx){
+  Logger.log('no col '+col_idx);
+  var count = 0;
+  var col_length = data.length;
+  for (var i = 0; i <col_length; i++){
+    if(typeof(data[i])==='undefined' || typeof(data[i][col_idx])==='undefined'){
+         return false;
+    }
+    if(typeof(data[i][col_idx])==='number'){
+          count++;   
+     }
```

Comment:
align one space to the left.. 

May be there is some plugins to format JS code according to google style guide, check if one is available